### PR TITLE
Feature #3188/import aligned eng ulb resource

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Added scripts for english USFM3 aligned data parse.
Updated en/ulb/tit to use aligned ULB, but rest of ULB content is unchanged.
Renamed ULB version to v12.1 (it's a beta)

---
This change is [Reviewable](https://reviewable.io/reviews/translationcoreapps/tc_resources/19)

### CODE REVIEW

**Comment by klappy** [comment]
Reviewed 3 of 2389 files at r1.
Review status: 3 of 1203 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
Reviewed 1 of 2389 files at r1.
Review status: 4 of 1203 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
Reviewed 3 of 2389 files at r1.
Review status: 7 of 1203 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
*[package.json, line 40 at r2](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19#-L5Epws0A0GlHrDP9sb3:-L5Epws0A0GlHrDP9sb4:bwxhwsh) ([raw file](https://github.com/translationcoreapps/tc_resources/blob/5313bbebcb4b0269b8e704e1763b6e3b6f46d494/package.json#L40)):*
> ```JSON
>     "fs-extra": "^3.0.1",
>     "path-extra": "3.0.0",
>     "usfm-js": "1.0.0-beta.24",
> ```

Can you update this to vbeta.25? We had to make one more release that doesn't include the __tests__ path in the release as it was over 100mb of text.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
Reviewed 2 of 4 files at r2.
Review status: 9 of 1203 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
Reviewed 2 of 4 files at r2.
Review status: 11 of 1203 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
*[helpers/ugntHelpers.js, line 35 at r2](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19#-L5ErXnvSqk2o4KGiGrR:-L5ErXnvSqk2o4KGiGrS:b-bxy0hj) ([raw file](https://github.com/translationcoreapps/tc_resources/blob/e3701d24245d4abca071bbed6fc6d049aba24f7a/helpers/ugntHelpers.js#L35)):*
> ```JavaScript
> }
> 
> /**
> ```

There's a lot of code removed here. Is it no longer needed or has it been moved?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
*[helpers/ugntHelpers.js, line 35 at r2](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19#-L5ErXnvSqk2o4KGiGrR:-L5ErsLJQZPufSsQ4d-L:b-a9dzgd) ([raw file](https://github.com/translationcoreapps/tc_resources/blob/e3701d24245d4abca071bbed6fc6d049aba24f7a/helpers/ugntHelpers.js#L35)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

There's a lot of code removed here. Is it no longer needed or has it been moved?
</blockquote></details>

OK. I see that they were moved to usfmHelpers. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by klappy** [comment]
Reviewed 2 of 2389 files at r1.
Review status: 13 of 1203 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

**Comment by mannycolon** [comment]
Reviewed 294 of 2389 files at r1.
Review status: 161 of 1203 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/translationcoreapps/tc_resources/19)*

File: helpers/englishUlbHelpers.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *                  UGNT.
+ */
+
+import path from 'path-extra';
+import * as bible from '../scripts/bible';
+import * as UsfmParseHelpers from './usfmParseHelpers';
+
+//TODO: EN_ULB_URL path is a temporary repo, needs to be updated to final location.
+
+const EN_ULB_URL = 'https://git.door43.org/photonomad0/AlignedUlb_en/raw/master';
```
Comment (mannycolon) [inline helpers/englishUlbHelpers.js]: @PhotoNomad0 Isn't the project supposed to come from the catalog?

File: __tests__/englishUlbHelpers.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      fs.ensureDir(UGNTOutputPath);
+      fs.copySync(resourceinputpath, UGNTOutputPath);
+
+      EnglishUlbHelpers.generateVersion(version, resolve, false);
+    }).then(() => {
+      console.log('ULB processing completed!');
+    });
+  }, 30000); // max timeout (should be long enough, but may need to be increased on a slow connection)
+
+  // it('should download and output en_ULB chapter files', () => {
```
Comment (mannycolon) [inline __tests__/englishUlbHelpers.test.js]: do we still need this code?

File: helpers/englishUlbHelpers.js (hunk 1 above)
Comment (PhotoNomad0) [reply helpers/englishUlbHelpers.js]: As marked in the TODO, this is temporary until the ULB has been aligned.

File: __tests__/englishUlbHelpers.test.js (hunk 2 above)
Comment (PhotoNomad0) [reply __tests__/englishUlbHelpers.test.js]: Left for TDD.

**Pull request merged**